addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiSmartLockHandler.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * @author Markus Katter - Initial contribution
+ */
+public class NukiSmartLockHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(NukiSmartLockHandler.class);
+
+    public NukiSmartLockHandler(Thing thing) {
+        super(thing);
+        logger.trace("Instantiating NukiSmartLockHandler({})", thing);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("NukiSmartLockHandler:initialize()");
+        Configuration bridgeConfiguration;
+        try {
+            bridgeConfiguration = getBridgeConfig();
+        } catch (NukiBridgeHandlerNotAvailableException e) {
+            logger.error("ERROR[{}]", e.getMessage(), e);
+            return;
+        }
+        String nukiId = (String) this.getConfig().get(NukiBindingConstants.CONFIG_NUKI_ID);
+        BridgeLockStateResponse bridgeLockStateResponse = new NukiHttpClient(bridgeConfiguration)
+                .getBridgeLockState(nukiId);
+        if (bridgeLockStateResponse.getStatus() == 200) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    bridgeLockStateResponse.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiSmartLockHandler:handleCommand({}, {})", channelUID, command);
+        Configuration bridgeConfiguration;
+        try {
+            bridgeConfiguration = getBridgeConfig();
+        } catch (NukiBridgeHandlerNotAvailableException e) {
+            logger.error("ERROR[{}]", e.getMessage(), e);
+            return;
+        }
+        String nukiId = (String) this.getConfig().get(NukiBindingConstants.CONFIG_NUKI_ID);
+        if (command instanceof RefreshType) {
+            handleCommandRefreshType(channelUID, command, bridgeConfiguration, nukiId);
+        } else if (command instanceof OnOffType) {
+            scheduler.execute(new Runnable() {
+                @Override
+                public void run() {
+                    handleCommandOnOffType(channelUID, command, bridgeConfiguration, nukiId);
+                }
+            });
+        } else {
+            logger.warn("NukiSmartLockHandler:handleCommand({}, {}) not implemented!", channelUID, command);
+        }
+    }
+
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.trace("NukiSmartLockHandler:handleUpdate({}, {})", channelUID, newState);
+        updateState(channelUID, newState);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiSmartLockHandler:dispose()");
+    }
+
+    private Configuration getBridgeConfig() throws NukiBridgeHandlerNotAvailableException {
+        logger.trace("Trying to getBridgeConfig for thing[{}]", this.getThing());
+        if (this.getBridge() != null && this.getBridge().getHandler() instanceof NukiBridgeHandler) {
```

Comment:
This does not work:
`
public NukiSmartLockHandler(Thing thing) {
        super(thing);
        Bridge bridge = getBridge();
}
`
bridge is null here. So I am not able to get the config here. Here I actually wanted to instantiate the NukiHttpClient one, and reuse it.